### DISCUSSION TITLE
Update your apps to the latest Firebase Cloud Messaging APIs and SDKs

### DISCUSSION THREAD

**Discussion by appsgenie** [invertase/react-native-firebase#7197] [General]
I've received an email from Firebase title `[Action Required] Update your apps to the latest Firebase Cloud Messaging APIs and SDKs` where it talks about that starting June 20, 2024 the legacy Firebase Cloud Messaging (FCM) APIs will be discontinued. The FAQ link was provided as [this ](https://notifications.google.com/g/p/ACUgQrGs1Rdh_Kqzx8reuQ-bLaLGWcMuEQMr8DXWN5oUI2JfBoSti-58oYaSQmLsRyEVf9Gv7kec-N-a_hmAgNtVFuddg2Edcmlm6f2tXUoFG3yUqXUAyR_nebumrXi7O9cOKZ8xKkzlN0pIB12w2jSCqefh0ho-ehSB521N4u-9SHet5IPYsDThZa8fDhygrONBckZun9H5UbE0-Gm12YJ8d3vPmFH4dFPhPw) and listing 2 APIs that seem to be affected on my project. 
Is there a way to know if the react-native-firebase package is affected by this as well and currently using any of the APIs that will be deprecated over the next year?

**Comment by mikehardy** [comment]
We continually update the underlying native SDKs that we use and offer all the native APIs for use.

The advice in the documentation you linked, related to the native SDKs looks like this:

![image](https://github.com/invertase/react-native-firebase/assets/782704/adbbabb6-d076-42f5-9360-28b2f95074e6)


We already use the latest SDKs, so there should be no action necessary related to react-native-firebase other than to stay current.

firebase-admin-sdk or other firebase SDKs (that are *not* the native iOS/android SDKs) are not in scope for this repository and you may have actions required there.

**Reply by appsgenie** [reply]
got it. so it might not even be the mobile client but admin sdk. I will check but I am not aware of using any FCM functionality hence was thinking maybe it's the phone auth inside react-native-firebase that uses silent notifications for device verifications..

**Comment by alikargar1993** [comment]
@mikehardy HI.
Currently, we are using `"@react-native-firebase/app": "14.9.3"` in our project, and for some limitations we can't update the Firebase version for now. Does this deprecation affect our Apps?

**Reply by mikehardy** [reply]
The message is vague so I'm not sure. As a maintainer that issues updates that fixes issues though, it doesn't look great to me when someone asks "I don't update to get your fixes, will this issue affect me?". I don't know? I issue new releases because answering questions like that takes effort that shouldn't be spent - effort should be spent moving forward. I'd plan the upgrade so you don't have to spend effort thinking about it.

There are little things too like "works with xcode 15" that are kind of important...

**Reply by posthello-code** [reply]
@alikargar1993 
These were the recommended FCM SDK versions I was recommended to use in an email from firebase.

```
Android
>= 23.1.2
iOS
>= 10.10.0
Web (Javascript)
>= 9.22.1
```

According to the changelog this means you need the latest version (v18) in order support iOS and Android
https://github.com/invertase/react-native-firebase/tree/main/CHANGELOG.md

**Comment by appsgenie** [comment]
For others who stumble upon this thread, and those who use Expo Notifications; if you are getting this notice and have `Legacy HTTP protocol` and `Server Keys` listed as impacted APIs, it could also be related to the Expo SDK instead of the react-native-firebase package.
At the time of this writing, the [docs](https://docs.expo.dev/push-notifications/push-notifications-setup/) say: `Expo Notifications only supports the Cloud Messaging API (Legacy) key at this time. This key is deprecated by Firebase. However, it will continue to work until June 20, 2024. We will provide information on migrating to the new v1 key in the future.`
